src/zope/schema/_bootstrapfields.py

```diff
@@ -664,6 +716,10 @@ def fromUnicode(self, value):
         finally:
             last_exc = None
 
+    # On Python 2, native strings are byte strings, which is
+    # what the converters expect, so we don't need to do any decoding.
+    fromBytes = fromUnicode if PY2 else lambda self, value: self.fromUnicode(value.decode('utf-8'))
```

Comment:
This would be more straightforward (and branch-testable) as its own method.

Reply:
Do you mean a whole method implementation, something like:
```python
def fromBytes(self, value):
   if PY2:
        return self.fromUnicode(...)
    else:
        return self.fromUnicode(value.decode(...))
```
?

I did originally have something like that, although it looked like this:
```python
if PY2:
    fromBytes = fromUnicode
else:
    def fromBytes(self, value):
        ...
```

I ran into two issues with that: first, coverage is run under python 3 locally, so that hurt coverage metrics, and I try really hard to avoid `#pragma: no cover` directives (this would be the case either way); second, that's an unnecessary method call under python 2, so we'd lose the potential optimisation this brings for users of lxml. 

I remain open to suggestions.